pr:pull-kubernetes-integration flaked 77 times in the past week

### Flaky Job: pr:pull-kubernetes-integration
 Flakes in the past week: **77**
 Consistency: **87.50%**

#### Flakiest tests by flake count:
| Test | Flake Count |
| --- | --- |
| k8s.io/kubernetes/test/integration/scheduler TestVolumeProvision | 31 |
| k8s.io/kubernetes/test/integration/evictions TestTerminalPodEviction | 23 |
| test-cmd run_rs_tests | 16 |

#### Previously closed issues for this job flaking:
#61384 #69857 


[Flakiest Jobs](https://storage.googleapis.com/k8s-metrics/flakes-latest.json)

OK so the test logs for integration tests are sliiiightly terrible.  I dug into the failure for the PR linked above.  The [text displayed in gubernator](https://gubernator.k8s.io/build/kubernetes-jenkins/pr-logs/pull/69675/pull-kubernetes-integration/33394#k8siokubernetestestintegrationscheduler-testvolumeprovision) (which incidentally is also what drives failure clustering in [go.k8s.io/triage](https://go.k8s.io/triage)) comes from [a junit.xml file](https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/69675/pull-kubernetes-integration/33383/artifacts/junit_f5a444384056ebac4f2929ce7b7920ea9733ca19_20181101-011119.xml)... but the failure message element appears to miss this crucial bit of text which is buried in [the corresponding junit.stdout file](https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/69675/pull-kubernetes-integration/33383/artifacts/junit_f5a444384056ebac4f2929ce7b7920ea9733ca19_20181101-011119.stdout)
```
--- FAIL: TestVolumeProvision (39.89s)
    volume_binding_test.go:807: Failed to schedule Pod "pod-i-pvc-prebound-w-provisioned": timed out waiting for the condition
```

Ah yes, it's our good friend `timed out waiting for condition`

--- a/pkg/scheduler/scheduler.go
+++ b/pkg/scheduler/scheduler.go
@@ -400,6 +400,10 @@ func (sched *Scheduler) bindVolumes(assumed *v1.Pod) error {
 			klog.Errorf("scheduler cache ForgetPod failed: %v", forgetErr)
 		}
 
+		// Volumes may be bound by PV controller asynchronously, we must clear
+		// stale pod binding cache.
+		sched.config.VolumeBinder.DeletePodBindings(assumed)
+
 		reason = "VolumeBindingFailed"
 		eventType = v1.EventTypeWarning
 		sched.config.Error(assumed, err)

